client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,392 @@
+                               new Argument("zoo", 123)
+                       ))),
+                       new Field("teamAffiliations", new HashSet<Field>(asList(
+                               new Field("name")))
+                       ))))
+       .addRootField("allHeroes", new HashSet<Field>(asList(new Field("name"))));
+
+----
+
+With the light API:
+[source,java]
+----
+Builder builderWithSugar = new Builder(Operation.Type.QUERY)
+       .addRootField("allHeroesIn",
+               asSet(
+                       arg("city", "New York, NY")
+               ),
+               asSet(
+                       field("name"),
+                       field("currentLocation", asSet(
+                               arg("foo", "bar"),
+                               arg("zoo", 123)
+                       )),
+                       field("teamAffiliations",
+                               asSet(
+                                       field("name")
+                               )
+                       )
+               )
+       )
+       .addRootField("allHeroes", asSet(field("name")));
+----
+
+== Building a mutation
+In this example, we want to add a hero to a team, more specifically Wolverine to the Avengers.
+
+The GraphQL mutation that responds to this need is:
+[source,json]
+----
+mutation {
+  addHeroToTeam(hero: "Wolverine", team: "Avengers") {
+         name
+         rivalTeam {
+             name
+	         members {
+		       costumeColor
+             }
+         }
+  }
+}
+----
+Note that there are 2 arguments on the addHeroToTeam field. The way to run it using the proposed API is:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.MUTATION)
+       .addRootField("addHeroToTeam",
+               asSet(
+                       arg("hero", "Wolverine"),
+                       arg("team", "Avengers")
+               ),
+               asSet(
+                       field("name"),
+                       field("rivalTeam",
+                               asSet(
+                                       field("name"),
+                                       field("members",
+                                               asSet(
+                                                       field("costumeColor")
+                                               )
+                                       )
+                               )
+                       )
+               )
+       )
+
+----
+
+== Managing input objects
+
+In that case, we want to create a hero from scratch:
+[source,json]
+----
+mutation {
+  createNewHero(hero: {
+	name: "Green Lantern",
+    	costumeColor: "green",
+    	dateOfLastCheckin: "2020-03-25",
+    	equipment: null,
+    	knownEnemies: [“Wolverine, “Starlord”, Iron Man”],
+	idNumber: 456) {
+    realName
+  }
+}
+----
+We can observe that we have a multi-valued argument and we need a specific API construct for that: the InputObject.For instance:
```

Comment:
Is `InputObject.For` actually the `new InputObject(...)` constructor or the `inputObject(...)` method?